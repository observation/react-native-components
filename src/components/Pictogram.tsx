import React from 'react'
import { View } from 'react-native'

import Pets from '../assets/pets.svg'
import Potplants from '../assets/potplants.svg'
import Selfies from '../assets/selfies.svg'
import Slash from '../assets/slash.svg'
import Log from '../lib/Log'
import theme from '../styles/theme'

type Props = {
  type: 'no_pets' | 'no_potplants' | 'no_selfies'
}

const width = 60
const height = 48

const Pictogram = ({ type }: Props) => {
  Log.debug('Pictogram', type)
  return (
    <View>
      {type === 'no_pets' && <Pets width={width} height={height} />}
      {type === 'no_potplants' && <Potplants width={width} height={height} />}
      {type === 'no_selfies' && <Selfies width={width} height={height} />}
      <Slash width={width} height={height} style={theme.absolute} />
    </View>
  )
}

export default Pictogram
