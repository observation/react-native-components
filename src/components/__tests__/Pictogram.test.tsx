import React from 'react'

import { render } from '@testing-library/react-native'

import Pictogram from '../Pictogram'

describe('Pictogram', () => {
  test('Rendering no pets', () => {
    const { toJSON } = render(<Pictogram type="no_pets" />)
    expect(toJSON()).toMatchSnapshot()
  })

  test('Rendering no potplants', () => {
    const { toJSON } = render(<Pictogram type="no_potplants" />)
    expect(toJSON()).toMatchSnapshot()
  })

  test('Rendering no selfies', () => {
    const { toJSON } = render(<Pictogram type="no_selfies" />)
    expect(toJSON()).toMatchSnapshot()
  })
})
