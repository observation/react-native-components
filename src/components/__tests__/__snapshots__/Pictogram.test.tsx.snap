// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pictogram Rendering no pets 1`] = `
<View>
  <pets.svg
    height={48}
    width={60}
  />
  <slash.svg
    height={48}
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    width={60}
  />
</View>
`;

exports[`Pictogram Rendering no potplants 1`] = `
<View>
  <potplants.svg
    height={48}
    width={60}
  />
  <slash.svg
    height={48}
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    width={60}
  />
</View>
`;

exports[`Pictogram Rendering no selfies 1`] = `
<View>
  <selfies.svg
    height={48}
    width={60}
  />
  <slash.svg
    height={48}
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    width={60}
  />
</View>
`;
